packages/core/src/pages/[slug]/p.tsx

```diff
@@ -39,19 +45,34 @@ const COMPONENTS: Record<string, ComponentType<any>> = {
   ...CUSTOM_COMPONENTS,
 }
 
-type Props = ServerProductPageQueryQuery &
-  PDPContentType & {
-    globalSections: GlobalSectionsData
-    meta: {
-      title: string
-      description: string
-      canonical: string
-    }
+type Props = PDPContentType & {
+  data: ServerProductPageQueryQuery
+  globalSections: GlobalSectionsData
+  meta: {
+    title: string
+    description: string
+    canonical: string
   }
+}
 
-function Page({ product, sections, globalSections, offers, meta }: Props) {
+const overwriteMerge = (_, sourceArray, __) => sourceArray
```

Comment:
Why is this needed? What does it do? Could you add a comment to the code pls?

```suggestion
const overwriteMerge = (_, sourceArray) => sourceArray
```

Reply:
this is the [Array Merge strategy from deepmerge](https://www.npmjs.com/package/deepmerge#:~:text=cloneUnlessOtherwiseSpecified(value%2C%20options)-,arrayMerge,-example%3A%20overwrite%20target). Without it we will have a lot of duplicated values that makes the pages didn't work as expected when we merge the values.
Comment added and unused arg removal [here](https://github.com/vtex/faststore/pull/1967/commits/16eedfc7b108394fe99a7e5c2bc63fa72ee365de).